### PR TITLE
lkl: Fix lkl_sys_halt and memory leak of TIF_HOST_THREAD

### PR DESCRIPTION
**Pull request by liuyuan10**
Signed-off-by: Yuan Liu <liuyuan@google.com>

---
This change is [Reviewable](https://reviewable.io/reviews/lkl/linux/262)

### CODE REVIEW

File: arch/lkl/kernel/setup.c [hunk 1]
```diff
@@ -112,7 +112,8 @@ void machine_restart(char *unused)
 long lkl_sys_halt(void)
 {
 	long err;
-	long params[6] = { 0, };
+	long params[6] = {LINUX_REBOOT_MAGIC1,
+		LINUX_REBOOT_MAGIC2, LINUX_REBOOT_CMD_RESTART, };
```
Comment (tavip) [inline arch/lkl/kernel/setup.c]: Not a big deal, but we probably want to use LINUX_REBOOT_CMD_HALT instead, just to match the API name.

Comment (liuyuan10) [inline arch/lkl/kernel/setup.c]: Looking at the kernel/reboot.c, LINUX_REBOOT_CMD_HALT will call do_exit which is not what we want.

**Comment by liuyuan10** [comment]
Added another fix for memory leak of TIF_HOST_THREAD on top of this PR

File: arch/lkl/kernel/setup.c (hunk 1 above)
Comment (tavip) [inline arch/lkl/kernel/setup.c]: Good point.

**Pull request merged**